flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/lsm/compact/ValueCountAccumulator.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.storage.filestore.lsm.compact;
+
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.storage.filestore.ValueKind;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+
+/**
+ * A {@link Accumulator} where key is the full record and value is a count which represents number
+ * of records of the exact same fields.
+ */
+public class ValueCountAccumulator implements Accumulator {
+
+    private long total;
+
+    @Override
+    public void reset() {
+        total = 0;
+    }
+
+    @Override
+    public void add(RowData value, ValueKind valueKind) {
+        long c = count(value);
+        switch (valueKind) {
+            case ADD:
+                total += c;
+                break;
+            case DELETE:
```

Comment:
The DELETE indicates that the value of the key should be delete. So we should assert here, never come in a DELETE value.